### DISCUSSION TITLE
Lesson 7: Where is the fulfillRandomness function executed ?

### DISCUSSION THREAD

**Discussion by LeCauchy** [smartcontractkit/full-blockchain-solidity-course-py#64] [Q&A] [answered]
Hello.
Each time I print the recentWinner, the output is : "0x0000000000000000000000000000000000000000 is the new winner!"
It seems like the function fulfillRandomness, which modify the recentWinner, is not executed. And I do not see where my code has a problem.

**Comment by LeCauchy** [comment, marked as answer]
I saw the problem. 
The time to get the randon number was greater than 60 seconds so when the message was printed, the recentWinner had not yet been modified. I got the good result on etherscan.

**Reply by PatrickAlphaC** [reply]
Nice work!
